File: go/utility/prepare_method.go
package utility

import "github.com/voxgig-sdk/bluefin-decryptx-p2pe-sdk/go/core"

func prepareMethodUtil(ctx *core.Context) string {
	opname := ctx.Op.Name

	methodMap := map[string]string{
		"create": "POST",
		"update": "PUT",
		"load":   "GET",
		"list":   "GET",
		"remove": "DELETE",
		"patch":  "PATCH",
	}

	if m, ok := methodMap[opname]; ok {
		return m
	}
	return "GET"
}

File: php/utility/PrepareMethod.php
<?php
declare(strict_types=1);

// BluefinDecryptxP2pe SDK utility: prepare_method

class BluefinDecryptxP2pePrepareMethod
{
    private const METHOD_MAP = [
        'create' => 'POST',
        'update' => 'PUT',
        'load' => 'GET',
        'list' => 'GET',
        'remove' => 'DELETE',
        'patch' => 'PATCH',
    ];

    public static function call(BluefinDecryptxP2peContext $ctx): string
    {
        return self::METHOD_MAP[$ctx->op->name] ?? 'GET';
    }
}
